retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/BufferSinkHttpData.java

```diff
@@ -163,7 +165,7 @@ public boolean isEndOfStream() {
 
     @Override
     public byte[] array() {
-        return bytes;
+        return buffer.readByteArray();
```

Comment:
Would it be possible for `BufferSinkHttpData` extend `Buffer` to save extra object allocation? `buffer()` could return `this`.